controllers/hetznerbaremetalremediation_controller_test.go

```diff
@@ -0,0 +1,428 @@
+		osSSHClientAfterInstallImage.On("CreateUserData", mock.Anything).Return(sshclient.Output{})
+		osSSHClientAfterInstallImage.On("EnsureCloudInit").Return(sshclient.Output{StdOut: "cloud-init"})
+		osSSHClientAfterInstallImage.On("CloudInitStatus").Return(sshclient.Output{StdOut: "status: done"})
+		osSSHClientAfterInstallImage.On("CheckCloudInitLogsForSigTerm").Return(sshclient.Output{})
+		osSSHClientAfterInstallImage.On("ResetKubeadm").Return(sshclient.Output{})
+		osSSHClientAfterInstallImage.On("GetHostName").Return(sshclient.Output{
+			StdOut: infrav1.BareMetalHostNamePrefix + hetznerBaremetalMachineName,
+			StdErr: "",
+			Err:    nil,
+		})
+		osSSHClientAfterCloudInit.On("Reboot").Return(sshclient.Output{})
+		osSSHClientAfterCloudInit.On("GetHostName").Return(sshclient.Output{
+			StdOut: infrav1.BareMetalHostNamePrefix + hetznerBaremetalMachineName,
+			StdErr: "",
+			Err:    nil,
+		})
+		osSSHClientAfterCloudInit.On("CloudInitStatus").Return(sshclient.Output{StdOut: "status: done"})
+		osSSHClientAfterCloudInit.On("CheckCloudInitLogsForSigTerm").Return(sshclient.Output{})
+		osSSHClientAfterCloudInit.On("ResetKubeadm").Return(sshclient.Output{})
+	})
+
+	AfterEach(func() {
+		Expect(testEnv.Cleanup(ctx, testNs, capiCluster, capiMachine, hetznerCluster)).To(Succeed())
+	})
+
+	Context("Basic test", func() {
+		Context("HetznerBareMetalHost will get provisioned", func() {
+			BeforeEach(func() {
+				hetznerSecret = getDefaultHetznerSecret(testNs.Name)
+				Expect(testEnv.Create(ctx, hetznerSecret)).To(Succeed())
+
+				rescueSSHSecret = helpers.GetDefaultSSHSecret("rescue-ssh-secret", testNs.Name)
+				Expect(testEnv.Create(ctx, rescueSSHSecret)).To(Succeed())
+
+				osSSHSecret = helpers.GetDefaultSSHSecret("os-ssh-secret", testNs.Name)
+				Expect(testEnv.Create(ctx, osSSHSecret)).To(Succeed())
+
+				bootstrapSecret = getDefaultBootstrapSecret(testNs.Name)
+				Expect(testEnv.Create(ctx, bootstrapSecret)).To(Succeed())
+			})
+
+			AfterEach(func() {
+				Expect(testEnv.Cleanup(ctx, testNs, hetznerSecret, osSSHSecret, rescueSSHSecret, bootstrapSecret)).To(Succeed())
+			})
+
+			Context("HetznerBaremetalHost doesn't exist", func() {
+				AfterEach(func() {
+					Expect(testEnv.Cleanup(ctx, hetznerBareMetalRemediation, hetznerBaremetalMachine)).To(Succeed())
+				})
+
+				It("should not remediate if no annotations is present in the hetznerBaremetalMachine", func() {
+					Expect(testEnv.Create(ctx, hetznerBaremetalMachine)).To(Succeed())
+					Expect(testEnv.Create(ctx, hetznerBareMetalRemediation)).To(Succeed())
+
+					Eventually(func() bool {
+						if err := testEnv.Get(ctx, capiMachineKey, capiMachine); err != nil {
+							return false
+						}
+
+						return isPresentAndFalseWithReason(capiMachineKey, capiMachine, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason)
+					}, timeout).Should(BeTrue())
+				})
+
+				It("should not remediate if no HetznerBareMetalHost is present", func() {
```

Comment:
... HetznerBareMetalHost does not exist anymore
this is more accurate, as we would otherwise not have that annotation